### PR TITLE
Ruby 3

### PR DESCRIPTION
**Pull request by JoeSouthan**
https://www.ruby-lang.org/en/news/2020/12/25/ruby-3-0-0-released/

- Webrick is no longer part of the Ruby 3 stdlib, added as a dependency
- Remove rubinus support
- Change `#enqueue` to remove the kwarg hash splat.

Due to the kwarg changes in Ruby 3 this causes the job arguments to be considered as part of the `*arg` splat. I've removed `#extract_attrs_and_args` to rely on kwargs for the job options and any other hash key is considered a job argument.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -21,11 +21,6 @@ group :test do
   gem 'rspec', '~> 3.9'
 end
 
-platforms :rbx do
-  gem 'json', '~> 1.8'
-  gem 'rubysl', '~> 2.0'
-end
-
```
Comment (vdandugc) [inline Gemfile:28]: possibly need to update documentation about removing support in readme doc.

**Comment by vdandugc** [comment]
looks good to me, don't see a [version change](https://github.com/gocardless/que/blob/master/lib/que/version.rb) , is this deliberate ?

**Review by vdandugc — APPROVED** [review]
just a few comments to address before merging.

**Comment by JoeSouthan** [comment]
@vdandugc I think we just use master for a lot of our projects

**Pull request merged**